rfcs/react-components/convergence/triggers-and-multiple-layers-of-button-handling.md

```diff
@@ -0,0 +1,124 @@
+
+Currently most triggers will ensure its direct children has the proper handler and properties of a button, by using `useARIAButtonProps` behind the scenes.
+
+In the case of a `div` as triggers child, `useARIAButtonProps` will ensure that this `div` will have proper `button` ARIA attributes
+
+```tsx
+<Trigger>
+  <div>Trigger something</div>
+</Trigger>
+```
+
+```html
+<div onclick="fn" onkeydown="fn" onkeyup="fn" role="button" tabindex="0">Trigger something</div>
+```
+
+`useARIAButtonProps` also ensure that on the case a `button` as trigger children, then no extra attributes are necessary.
+
+```tsx
+<Trigger>
+  <button>Trigger something</button>
+</Trigger>
+```
+
+```html
+<button>Trigger something</button>
+```
+
+## Problem statement
+
+On the case of a custom component on the other hand, it's impossible for `useARIAButtonProps` to find out if the returned element will be an actual button or not.
+
+```tsx
+<Trigger>
+  <Button>Trigger something</Button>
+</Trigger>
+```
+
+That way, even for `Button`, `useARIAButtonProps` will end up providing a bunch of unnecessary props.
+
+```html
+<button onclick="fn" onkeydown="fn" onkeyup="fn" role="button" tabindex="0">Trigger something</button>
+```
+
+## Detailed Design or Proposal
+
+### Mark `Button` component
+
+The Trigger API has a type called [FluentTriggerComponent](https://github.com/microsoft/fluentui/blob/92bc886e6e3e5ec43847752941456666eb69d32b/packages/react-components/react-utilities/src/trigger/types.ts#L15-L17), this type is used to mark every Trigger compliant component as a Trigger, take [MenuTrigger](https://github.com/microsoft/fluentui/blob/92bc886e6e3e5ec43847752941456666eb69d32b/packages/react-components/react-menu/src/components/MenuTrigger/MenuTrigger.tsx#L7-L18) as an example. This was required by the Trigger API to ensure traversing down multiple triggers:
+
+```tsx
+<Tooltip>
+  <MenuTrigger>
+    <button>trigger</button>
+  </MenuTrigger>
+</Tooltip>
+```
+
+By providing something like `ARIAButtonComponent`, we could mark `Button` component the same way trigger components are marked, but in this case to ensure `useARIAButtonProps` are not unnecessarily reused.
+
+```tsx
+/**
+ * Buttons give people a way to trigger an action.
+ */
+export const Button: ForwardRefComponent<ButtonProps> & ARIAButtonComponent = React.forwardRef((props, ref) => {
+  const state = useButton_unstable(props, ref);
+
+  useButtonStyles_unstable(state);
+
+  return renderButton_unstable(state);
+});
+
+Button.displayName = 'Button';
+Button.isARIAButtonComponent = true; // this will ensure that `useARIAButtonProps` doesn't repeat itself
+```
+
+Then inside `useARIAButtonProps`:
+
+```tsx
+// if its a marked component, than do nothing, just return original props
+if (isARIAButtonComponent(element)) {
+  return props;
+}
+```
+
+#### Pros
+
+1. Straight forward solution, `Button` will work with triggers out of the box
+
+#### Cons
+
+1. won't work with `CustomButton` wrapping `Button`
```

Comment:
This is the biggest issue as it works against our composition model and composition model in React as requires to know about `isFluentTriggerComponent`. It works for `MenuTrigger`, `PopoverTrigger` more or less because there are no reasons to recompose `*Trigger` components.